riotfile.py

```diff
@@ -187,16 +189,22 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
         ),
         Venv(
             name="tracer",
-            command="pytest {cmdargs} tests/tracer/",
+            pkgs={
+                "msgpack": latest,
+                "attrs": ["==19.2.0", latest],
+                "packaging": ["==17.1", latest],
+            },
             venvs=[
+                # pytest-xdist collects different tests in different processes,
+                # and fails on Python 3.5.
                 Venv(
-                    pys=select_pys(),
-                    pkgs={
-                        "msgpack": latest,
-                        "attrs": ["==19.2.0", latest],
-                        "packaging": ["==17.1", latest],
-                    },
-                )
+                    pys="3.5",
+                    command="pytest {cmdargs} tests/tracer/",
+                ),
+                Venv(
+                    pys=select_pys(max_version="3") + select_pys(min_version="3.6"),
+                    command="pytest -n 2 --dist loadscope {cmdargs} tests/tracer/",
```

Comment:
I wonder if we should do a global env variable for the parallelism? that way maybe locally it is 2 by default, but on CircleCI we might be able to say something higher

Reply:
In earlier attempts I had used `auto` and that started 36 workers, but I doubt CircleCI give us 36 physical cores, so I'd expect this to actually starve the few cores available and cause overheads. Hence the more conservative approach of manually requesting just 2 parallel workers.